### PR TITLE
[i18nIgnore] Fix various typos in `src/content/docs/en/guides`

### PR DESCRIPTION
**Pull request by anaxite**
#### Description

This PR fixes various high-profile typos in the English _Guides_ section that don't seem to have an i18n impact:

- Brand capitalization
- Duplicate words
- Missing letters
- A few consistency checks
- etc.

### CODE REVIEW

File: src/content/docs/en/guides/astro-db.mdx [hunk 1]
```diff
@@ -176,7 +176,7 @@ The steps below will guide you through the process of installing the Turso CLI,
 
 1. Install the [Turso CLI](https://docs.turso.tech/cli/installation).
 
-2. [Login or signup](https://docs.turso.tech/cli/authentication) to Turso.
+2. [Login or sign up](https://docs.turso.tech/cli/authentication) to Turso.
```
Comment (sarah11918) [inline src/content/docs/en/guides/astro-db.mdx:179]:
```suggestion
2. [Log in or sign up](https://docs.turso.tech/cli/authentication) to Turso.
```
If we're doing this... 😄

File: src/content/docs/en/guides/backend/xata.mdx [hunk 2]
```diff
@@ -62,7 +62,7 @@ interface ImportMeta {
 Read more about [environment variables](/en/guides/environment-variables/) and `.env` files in Astro.
 :::
 
-Using the codegeneration from the Xata CLI and choosing the TypeScript option, generated an instance of the SDK for you, with types tailored to your database schema. Additionally, `@xata.io/client` was added to your `package.json`.
+Using the code generation from the Xata CLI and choosing the TypeScript option, generated an instance of the SDK for you, with types tailored to your database schema. Additionally, `@xata.io/client` was added to your `package.json`.
```
Comment (sarah11918) [inline src/content/docs/en/guides/backend/xata.mdx:65]: Just checking that you're sure this isn't some proper name/term!

**Review by sarah11918 — COMMENTED** [review]
Thanks for this assistance, @anaxite ! Left just a few quick things to check on!

File: src/content/docs/en/guides/backend/xata.mdx (hunk 2 above)
Comment (sarah11918) [reply src/content/docs/en/guides/backend/xata.mdx:65]: Just checked on this, and looks like it's the `codegen` CLI command, so yes, this should just be in regular English words!

**Review by sarah11918 — APPROVED** [review]
OK, good to go! :muscle:

**Review by yanthomasdev — APPROVED** [review]
LGTM! Thanks!

File: src/content/docs/en/guides/cms/drupal.mdx [hunk 3]
```diff
@@ -103,7 +103,7 @@ JSON:API requests and responses can often be complex and deeply nested. To simpl
 
 Your content is fetched from a JSON:API URL.
 
-### Drupal JSON:API URL Sructure
+### Drupal JSON:API URL Structure
```
[the remark this answers is not in the document]
Comment (anaxite) [reply src/content/docs/en/guides/cms/drupal.mdx]: Thank you! (I appreciate the case as well)

**Pull request merged**